.github/workflows/ci.yml

```diff
@@ -31,8 +31,12 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install Poetry
         uses: snok/install-poetry@v1
+      - name: Generate lock file
+        run: poetry lock
+      - name: Update lock file
+        run: poetry update
```

Comment:
```suggestion
```
The lockfile should not be automatically updated by our CI job, but rather locally - else we lose the benefits of locked dependencies from poetry.

Best update the lockfile locally - without updating any of the other dependencies with `poetry lock --no-update`